Performance Benchmarks Workflow Failing Due to Syntax Error

@claude 

# Performance Benchmarks Workflow Failing Due to Syntax Error

## Summary
The Performance Benchmarks GitHub Actions workflow is failing due to a syntax error in the test file `test_suricata_performance.py`. This prevents the benchmark tests from running and collecting performance data.

## Error Details
**Location:** `tests/performance_tests/test_suricata_performance.py`, line 44  
**Error Type:** `SyntaxError: EOL while scanning string literal`  
**Problematic Code:** `f"Benchmark: {`

## Full Error Trace
```
File "/home/runner/work/zeek-yara-integration/zeek-yara-integration/tests/performance_tests/test_suricata_performance.py", line 44
E       f"Benchmark: {
E                     ^
E   SyntaxError: EOL while scanning string literal
```

## Impact
- Performance benchmarks cannot be executed
- No benchmark data is being collected (`Not saving anything, no benchmarks have been run!`)
- Workflow exits with code 2, failing the CI/CD pipeline

## Environment
- **OS:** Ubuntu 24.04.2 LTS
- **Python:** 3.9.23
- **pytest:** 8.4.1
- **pytest-benchmark:** 5.1.0
- **Workflow:** Performance Benchmarks

## Expected Behavior
The workflow should successfully run performance tests and generate benchmark data in JSON format.

## Steps to Reproduce
1. Trigger the Performance Benchmarks workflow
2. Observe the pytest collection phase failing with syntax error
3. Workflow terminates without running any benchmarks

## Suggested Fix
The issue appears to be an incomplete f-string on line 44 of `test_suricata_performance.py`. The f-string needs to be properly closed:

**Current (broken):**
```python
f"Benchmark: {
```

**Should be something like:**
```python
f"Benchmark: {some_variable}"
```

## Action Required
1. Review line 44 in `tests/performance_tests/test_suricata_performance.py`
2. Complete the f-string expression
3. Ensure proper syntax validation before committing
4. Re-run the Performance Benchmarks workflow to verify the fix

## Priority
**High** - This is blocking the performance testing pipeline and preventing benchmark data collection.

## Labels
- `bug`
- `ci/cd`
- `performance`
- `testing`
- `syntax-error`

--- a/tests/performance_tests/test_suricata_performance.py
+++ b/tests/performance_tests/test_suricata_performance.py
@@ -41,11 +41,7 @@ except ImportError:
                     result = func(*args, **kwargs)
                 total_time = time.time() - start_time
                 print(
-                    f"Benchmark: {
-                        func.__name__} - Total time: {
-                        total_time:.6f}s, Average: {
-                        total_time /
-                        iterations:.6f}s"
+                    f"Benchmark: {func.__name__} - Total time: {total_time:.6f}s, Average: {total_time / iterations:.6f}s"
                 )
                 return result
 
@@ -330,16 +326,13 @@ class TestSuricataPerformance:
         # Log performance metrics
         print(f"Retrieved {len(all_alerts)} alerts in {duration:.6f} seconds")
         print(
-            f"Average time per alert: {
-                duration /
-                len(all_alerts):.6f} seconds"
+            f"Average time per alert: {duration / len(all_alerts):.6f} seconds"
         )
 
         # Test should run in reasonable time (adjust threshold as needed)
         assert (
             duration < 5.0
-        ), f"Alert retrieval took {
-            duration:.6f} seconds, which exceeds the 5-second threshold"
+        ), f"Alert retrieval took {duration:.6f} seconds, which exceeds the 5-second threshold"
 
     def test_alert_filtering_performance(self, performance_test_env, timer):
         """Test performance of filtered alert retrieval"""
@@ -365,16 +358,13 @@ class TestSuricataPerformance:
 
             # Log performance metrics
             print(
-                f"Filter {filter_dict}: Retrieved {
-                    len(filtered_alerts)} alerts in {
-                    duration:.6f} seconds"
+                f"Filter {filter_dict}: Retrieved {len(filtered_alerts)} alerts in {duration:.6f} seconds"
             )
 
             # Test should run in reasonable time
             assert (
                 duration < 1.0
-            ), f"Filtered retrieval took {
-                duration:.6f} seconds, which exceeds the 1-second threshold"
+            ), f"Filtered retrieval took {duration:.6f} seconds, which exceeds the 1-second threshold"
 
     @benchmark(iterations=5)
     def test_correlation_performance(self, performance_test_env):
@@ -403,10 +393,7 @@ class TestSuricataPerformance:
 
             # Log performance metrics
             print(
-                f"Correlated {
-                    alert_counts['yara_count']} YARA and {
-                    alert_counts['suricata_count']} Suricata alerts in {
-                    duration:.6f} seconds"
+                f"Correlated {alert_counts['yara_count']} YARA and {alert_counts['suricata_count']} Suricata alerts in {duration:.6f} seconds"
             )
             print(f"Found {len(correlated_groups)} correlation groups")
 
@@ -467,14 +454,10 @@ class TestSuricataPerformance:
 
         # Log performance metrics
         print(
-            f"Stored {
-                len(correlation_groups)} correlation groups in {
-                duration:.6f} seconds"
+            f"Stored {len(correlation_groups)} correlation groups in {duration:.6f} seconds"
         )
         print(
-            f"Average time per group: {
-                duration /
-                len(correlation_groups):.6f} seconds"
+            f"Average time per group: {duration / len(correlation_groups):.6f} seconds"
         )
 
         # Verify correlations were stored
@@ -489,8 +472,7 @@ class TestSuricataPerformance:
         # Test should run in reasonable time
         assert (
             duration < 10.0
-        ), f"Correlation storage took {
-            duration:.6f} seconds, which exceeds the 10-second threshold"
+        ), f"Correlation storage took {duration:.6f} seconds, which exceeds the 10-second threshold"
 
     def test_process_alerts_performance(self, performance_test_env, timer):
         """Test performance of alert processing"""
@@ -548,5 +530,4 @@ class TestSuricataPerformance:
         # Test should run in reasonable time
         assert (
             duration < 10.0
-        ), f"Alert processing took {
-            duration:.6f} seconds, which exceeds the 10-second threshold"
+        ), f"Alert processing took {duration:.6f} seconds, which exceeds the 10-second threshold"

